### PR TITLE
Signed unsigned

### PR DESCRIPTION
**Pull request by mbouron**

### CODE REVIEW

File: libnopegl/src/api.c [hunk 1]
```diff
@@ -141,8 +141,8 @@ static const char *get_cap_string_id(unsigned cap_id)
 
 static int load_caps(struct ngl_backend *backend, const struct gpu_ctx *gpu_ctx)
 {
-    const int has_compute        = NGLI_HAS_ALL_FLAGS(gpu_ctx->features, NGLI_FEATURE_COMPUTE);
-    const int has_ds_resolve     = NGLI_HAS_ALL_FLAGS(gpu_ctx->features, NGLI_FEATURE_DEPTH_STENCIL_RESOLVE);
+    const uint32_t has_compute        = NGLI_HAS_ALL_FLAGS(gpu_ctx->features, NGLI_FEATURE_COMPUTE);
+    const uint32_t has_ds_resolve     = NGLI_HAS_ALL_FLAGS(gpu_ctx->features, NGLI_FEATURE_DEPTH_STENCIL_RESOLVE);
```
Comment (ubitux) [inline libnopegl/src/api.c:145]: These are boolean so `int` is appropriate; I'd rather cast explicitly below when fit into their u32 container

File: libnopegl/src/backends/gl/rendertarget_gl.c [hunk 2]
```diff
@@ -94,15 +94,15 @@ static int create_fbo(struct rendertarget *s, int resolve, GLuint *idp)
...
+    uint32_t nb_color_attachments = 0;
 
     ngli_glGenFramebuffers(gl, 1, &id);
     ngli_glBindFramebuffer(gl, GL_FRAMEBUFFER, id);
 
     for (size_t i = 0; i < params->nb_colors; i++) {
         const struct attachment *attachment = &params->colors[i];
         const struct texture *texture = resolve ? attachment->resolve_target : attachment->attachment;
-        const int layer = resolve ? attachment->resolve_target_layer : attachment->attachment_layer;
+        const uint32_t layer = resolve ? attachment->resolve_target_layer : attachment->attachment_layer;
```
Comment (ubitux) [inline libnopegl/src/backends/gl/rendertarget_gl.c:105]: the gl functions takes an `int` for the layer

File: libnopegl/src/node_rtt.c [hunk 3]
```diff
@@ -278,8 +278,8 @@ static int rtt_prefetch(struct ngl_node *node)
         const struct rtt_texture_info info = get_rtt_texture_info(o->color_textures[i]);
         struct texture_priv *texture_priv = info.texture_priv;
         struct texture *texture = texture_priv->texture;
-        const int32_t layer_end = info.layer_base + info.layer_count;
-        for (int32_t j = info.layer_base; j < layer_end; j++) {
+        const uint32_t layer_end = info.layer_base + info.layer_count;
+        for (uint32_t j = info.layer_base; j < layer_end; j++) {
             s->rtt_params.colors[s->rtt_params.nb_colors++] = (struct attachment) {
                 .attachment       = texture,
                 .attachment_layer = j,
```
Comment (ubitux) [inline libnopegl/src/node_rtt.c:285]: `attachment_layer` is i32

File: libnopegl/src/rendertarget.h [hunk 4]
```diff
@@ -51,9 +51,9 @@ struct rendertarget_layout {
 
 struct attachment {
     struct texture *attachment;
-    int32_t attachment_layer;
+    uint32_t attachment_layer;
     struct texture *resolve_target;
-    int32_t resolve_target_layer;
+    uint32_t resolve_target_layer;
```
Comment (ubitux) [inline libnopegl/src/rendertarget.h:56]: this is assigned some `int` in `rtt.c`

File: libnopegl/src/api.c (hunk 1 above)
Comment (mbouron) [reply libnopegl/src/api.c]: Commit dropped.

File: libnopegl/src/backends/gl/rendertarget_gl.c (hunk 2 above)
Comment (mbouron) [reply libnopegl/src/backends/gl/rendertarget_gl.c]: Commit dropped.

File: libnopegl/src/node_rtt.c (hunk 3 above)
Comment (mbouron) [reply libnopegl/src/node_rtt.c]: Commit dropped.

File: libnopegl/src/rendertarget.h (hunk 4 above)
Comment (mbouron) [reply libnopegl/src/rendertarget.h]: Commit dropped.

**Review by ubitux — APPROVED** [review]

**Pull request merged**